File: (03-29-2024)Gas Station/joshlin/solution.py
from typing import List

class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        total, diff, start = 0, 0, 0
        for i in range(len(gas)):
            total += gas[i] - cost[i]
            diff += gas[i] - cost[i]
            if diff < 0:
                diff = 0
                start = i + 1
        return -1 if total < 0 else start

File: (03-29-2024)Gas Station/joshlin/solution.ts
function canCompleteCircuit(gas: number[], cost: number[]): number {
  let [total, diff, start] = [0, 0, 0];
  for (let i = 0; i < gas.length; i++) {
      total += gas[i] - cost[i];
      diff += gas[i] - cost[i];
      if (diff < 0) {
          diff = 0;
          start = i + 1;
      }
  }
  return total >= 0 ? start : -1;
};